shoop/themes/classic_gray/templates/classic_gray/shoop/front/product/detail.jinja

```diff
@@ -19,9 +19,12 @@
         <li><a href="/">{% trans %}Home{% endtrans %}</a></li>
         {% set category = shop_product.primary_category %}
         {% if category %}
-            {% for c in category.get_ancestors() if c.is_visible(customer=request.customer) %}
-                <li><a href="{{ shoop.urls.model_url(c) }}">{{ c.name }}</a></li>
+            {% for child_cat in category.get_ancestors() if child_cat.is_visible(customer=request.customer) %}
```

Comment:
You can pass `include_self=True` to `get_ancestors` to simplify the loop further.